cmake/GenConfig.cmake

```diff
@@ -0,0 +1,132 @@
+#  set(USE_IPPIW TRUE)
+#  file(RELATIVE_PATH IPPIW_INSTALL_PATH_RELATIVE_CONFIGCMAKE "${CMAKE_BINARY_DIR}" "${IPPIW_LOCATION_PATH}")
+#  ocv_cmake_configure("${CMAKE_CURRENT_LIST_DIR}/templates/VISPConfig-IPPIW.cmake.in" IPPIW_CONFIGCMAKE @ONLY)
+#else()
+#  set(USE_IPPIW FALSE)
+#endif()
+
+configure_file("${VISP_SOURCE_DIR}/cmake/templates/VISPConfig.cmake.in" "${CMAKE_BINARY_DIR}/VISPConfig.cmake" @ONLY)
+#support for version checking when finding opencv. find_package(OpenCV 2.3.1 EXACT) should now work.
+configure_file("${VISP_SOURCE_DIR}/cmake/templates/VISPConfig-version.cmake.in" "${CMAKE_BINARY_DIR}/VISPConfig-version.cmake" @ONLY)
+
+# --------------------------------------------------------------------------------------------
+#  Part 2/3: ${BIN_DIR}/unix-install/VISPConfig.cmake -> For use *with* "make install"
+# -------------------------------------------------------------------------------------------
+file(RELATIVE_PATH VISP_INSTALL_PATH_RELATIVE_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/${VISP_CONFIG_INSTALL_PATH}/" ${CMAKE_INSTALL_PREFIX})
+set(VISP_INCLUDE_DIRS_CONFIGCMAKE "\"\${VISP_INSTALL_PATH}/${VISP_INCLUDE_INSTALL_PATH}\" \"\${VISP_INSTALL_PATH}/${VISP_INCLUDE_INSTALL_PATH}/VISP\"")
+
+
+# TODO: not using IPPICV
+#if(USE_IPPICV)
+#  file(RELATIVE_PATH IPPICV_INSTALL_PATH_RELATIVE_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}" "${IPPICV_INSTALL_PATH}")
+#  ocv_cmake_configure("${CMAKE_CURRENT_LIST_DIR}/templates/VISPConfig-IPPICV.cmake.in" IPPICV_CONFIGCMAKE @ONLY)
+#endif()
+#if(USE_IPPIW)
+#  file(RELATIVE_PATH IPPIW_INSTALL_PATH_RELATIVE_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}" "${IPPIW_INSTALL_PATH}")
+#  ocv_cmake_configure("${CMAKE_CURRENT_LIST_DIR}/templates/VISPConfig-IPPIW.cmake.in" IPPIW_CONFIGCMAKE @ONLY)
+#endif()
+
+function(vp_gen_config TMP_DIR NESTED_PATH ROOT_NAME)
+  vp_path_join(__install_nested "${VISP_CONFIG_INSTALL_PATH}" "${NESTED_PATH}")
+  vp_path_join(__tmp_nested "${TMP_DIR}" "${NESTED_PATH}")
+
+  file(RELATIVE_PATH VISP_INSTALL_PATH_RELATIVE_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/${__install_nested}" "${CMAKE_INSTALL_PREFIX}/")
+
+  configure_file("${VISP_SOURCE_DIR}/cmake/templates/VISPConfig-version.cmake.in" "${TMP_DIR}/VISPConfig-version.cmake" @ONLY)
+
+  configure_file("${VISP_SOURCE_DIR}/cmake/templates/VISPConfig.cmake.in" "${__tmp_nested}/VISPConfig.cmake" @ONLY)
+  install(EXPORT VISPModules DESTINATION "${__install_nested}" FILE VISPModule.cmake COMPONENT dev)
+  install(FILES
+      "${TMP_DIR}/VISPConfig-version.cmake"
+      "${__tmp_nested}/VISPConfig.cmake"
+      DESTINATION "${__install_nested}" COMPONENT dev)
+
+  if(ROOT_NAME)
+    # Root config file
+    configure_file("${VISP_SOURCE_DIR}/cmake/templates/${ROOT_NAME}" "${TMP_DIR}/VISPConfig.cmake" @ONLY)
+    install(FILES
+        "${TMP_DIR}/VISPConfig-version.cmake"
+        "${TMP_DIR}/VISPConfig.cmake"
+        DESTINATION "${VISP_CONFIG_INSTALL_PATH}" COMPONENT dev)
+  endif()
+endfunction()
+
```

Comment:
I see more this function in `cmake/VISPUtils.cmake`

Reply:
transferred the function to VISPUtils.cmake